[21기_최근호] Docker를 이용한 배포 미션 제출합니다.

WithFortuna opened the pull request.

WithFortuna committed:
docs: README 작성

WithFortuna commented:
배포 중 계속되는 버그를 만났습니다.. 서둘러서 커밋 추가하겠습니다....!

WithFortuna committed:
[CHORE] Dockerfile 포트 명시

포트를 명시했습니다

WithFortuna committed:
[DOCS] README용 이미지 업로드

WithFortuna committed:
[DOCS] README용 이미지 업로드

WithFortuna committed:
[DOCS] README 작성

WithFortuna commented:
AppRunner의 장점은 간편한 배포인데  써도 rds, elastic cache까지도 별도로 사용해야하다보니 장점이 사라지고 오히려 복잡해지네요... 아키텍처가 커질수록 AppRunner는 비추라고 생각되는데, 리뷰어님들 의견 한번 듣고싶습니다..

SeoyeonPark1223 commented on Dockerfile:
```diff
@@ -1,8 +1,9 @@
-ENV SPRING_PROFILES_ACTIVE=docker
+EXPOSE 8080
+ENV SPRING_PROFILES_ACTIVE=prod
 ENV JWT_SECRET_KEY=4iLaz4RBs9F0g1WAVppByh0uH/oeT6yT0zWzLx8fGPE
 ENV JWT_REFRESH_TOKEN_EXPIRATION=86400000
 ENV JWT_ACCESS_TOKEN_EXPIRATION=60000
```
이 부분은 빌드 단계에서 굳이 필요가 없는 값이니 이렇게 노출하기보다는 더미로 넣어주고 런 단계에서 환경변수를 넘겨주는건 어떨까요?!

WithFortuna replied:
! 도커랑 배포 상에서도 환경변수 관리하는법을 공부해봐야겠습니다

SeoyeonPark1223 commented on README.md:
```diff
@@ -390,3 +390,175 @@ Docker는 애플리케이션을 실행 환경과 함께 **컨테이너**라는 
+## 오류 발생!!
+- 상황: 회원가입은 통과, But 로그인은 500오류
+- **추정 원인**: Redis, DB 주소
+- **해결 방법**: RDS, ElasticCache 사용 필요
```
으아... 앱 러너는 Redis 캐시 서버 이미지를 제공해주지 않는군요.... 너무 힘들었을거같아요

WithFortuna replied:
맞아유.. 단일 컨테이너의 실행만 가능합니다

SeoyeonPark1223 commented on README.md:
```diff
@@ -390,3 +390,175 @@ Docker는 애플리케이션을 실행 환경과 함께 **컨테이너**라는 
+
+## 정리 메모
+
+- App Runner가 VPC 자원에 접근하려면 반드시 VPC Connector가 필요함
+- Redis 사용 시 다음 사항 확인 필요
+
+  - TLS 설정 여부 (`--tls`)
+  - 보안 그룹 Inbound 규칙
+  - application-prod.yml 내 SSL 설정 여부
```
깔끔한 정리입니다!

SeoyeonPark1223 commented on README.md:
```diff
@@ -390,3 +390,175 @@ Docker는 애플리케이션을 실행 환경과 함께 **컨테이너**라는 
+  - 보안 그룹 Inbound 규칙
+  - application-prod.yml 내 SSL 설정 여부
+
+### 배포 api 테스트 성공
```
그래서 app runner 사용과 ec2+docker 사용은 장단점이 명확한 것 같네요 찾아보니 앱 러너를 사용하면 이후 자동 배포도 GitHub Actions 사용하지 않고도 할 수 있고 https도 기본 지원해주고 이래저래 편한 것 같아요 하지만.. 생각보다 여러 리소소를 많이 잡아먹고 설정이 매우 복잡한 것 같아요 
그래도 뭔가 실무로 가면 근호님이 이렇게 트러블슈팅과 과정을 공부해둔게 많이 도움될 것 같슴다!!

WithFortuna replied:
아키텍처가 정말 단순한 경우엔 쓸만해보입니다!

SeoyeonPark1223 reviewed (commented):
이번주차 과제도 수고많으셨습니다~!

WithFortuna committed:
[DOCS] gitignore 수정

WithFortuna committed:
[CHORE] 환경변수 관리

 - DB 경로 환경변수로 관리
 - local환경은 환경변수를 .env 파일에서 주입하도록 변경

WithFortuna committed:
[CHORE] DockerFile 변경

기존: 로컬환경에서 빌드를 하고 docker에 jar를 올림
변경:
- docker 안에서 빌드를 함(의존성이 변경되지 않으면 캐싱 가능)
- alpine 사용: image용량도 감소(디버깅엔 어려움이 있음)

WithFortuna committed:
[STYLE] 주석 추가 및 삭제

WithFortuna committed:
[STYLE] 주석 추가 및 삭제

WithFortuna committed:
Revert "[STYLE] 주석 추가 및 삭제"

This reverts commit 8b92c7ec621a4f19bae02de57b4ee220ca6d7bd6.

WithFortuna committed:
[REFACTOR] 테스트 프로파일 명시 삭제

명시하지 않고 환경변수로 관리함

WithFortuna committed:
Merge branch 'master' of github.com:WithFortuna/spring-knowledgeIn-21st

# Conflicts:
#	Dockerfile

leerura commented on Dockerfile:
```diff
@@ -1,8 +1,34 @@
+# => 이렇게 하니까 image용량이 1/4로 줄음(884MB -> 229MB)
+#############################################
+# 1) Build Stage – Gradle + JDK17 (Alpine)  #
+#############################################
```
오호 신기하네용

leerura commented on README.md:
```diff
@@ -390,3 +390,175 @@ Docker는 애플리케이션을 실행 환경과 함께 **컨테이너**라는 
+## 가능한 구현 방법
+- ECR + App Runner
+- ECR + ECS
+
```
ECR이랑 App Runner 처음 보는데 덕분에 공부하고 갑니당

leerura commented on README.md:
````diff
@@ -390,3 +390,175 @@ Docker는 애플리케이션을 실행 환경과 함께 **컨테이너**라는 
+```
+
+- 이거 안넣어서 1시간 헤맸습니다.... .
+
````
문제 해결 과정 써주신 게 인상 깊네요!

WithFortuna replied:
감사합니담~~
